lib/fibonacci.rb

```diff
@@ -1,9 +1,27 @@
 # Improved Fibonacci
 
-# Time Complexity - ?
-# Space Complexity - ?  (should be O(n))
+# Time Complexity - O(n)
+# Space Complexity - O(n)  (should be O(n))
 # Hint, you may want a recursive helper method
 # Hint:  Fib(0) = 0, Fib(1) = 1, work up from there
+
 def fibonacci(n)
```

Comment:
This works, but can you do it without maintaining a list of Fibonacci solutions from 0..n?